node/src/genesis.rs

```diff
@@ -1,26 +1,298 @@
+                }
+            },
+            Err(err) => {
+                return Err(anyhow!(
+                    "Failed to generate new genesis file \"{}\". Error: {err}",
+                    output_path.display()
+                ));
+            },
+        }
+    }
+
+    let genesis_file_path = Path::new(config_filepath);
+
+    if let Ok(file_exists) = genesis_file_path.try_exists() {
+        if !file_exists {
+            return Err(anyhow!(
+                "The {} file does not exist. It is necessary to initialise the node GenesisState.",
+                genesis_file_path.display()
+            ));
+        }
+    } else {
+        return Err(anyhow!("Failed to open process {} file.", genesis_file_path.display()));
+    }
+
+    let genesis_input: GenesisInput = load_config(genesis_file_path).extract().map_err(|err| {
+        anyhow!("Failed to load {} config file: {err}", genesis_file_path.display())
+    })?;
+    println!("Config file: {} has successfully been loaded.", genesis_file_path.display());
+
+    let accounts = create_accounts(&genesis_input.accounts)?;
+    println!("Accounts have successfully been created.");
+
+    let genesis_state = GenesisState::new(accounts, genesis_input.version, genesis_input.timestamp);
+
+    // Write genesis state as binary format
+    fs::write(output_path, genesis_state.to_bytes()).unwrap_or_else(|_| {
+        panic!("Failed to write genesis state to output file {}", output_path.display())
+    });
+
+    println!("Genesis initialisation successful: {} has been created", output_path.display());
+
+    Ok(())
+}
+
+fn create_account_file(
+    account: &Account,
+    seed: Word,
+    auth_scheme: String,
+    auth_seed: String,
+    index: usize,
+) -> Result<()> {
+    let file_path = PathBuf::from(format!("accounts/account{index}.mac"));
+
+    let account_data = AccountData {
+        account: account.clone(),
+        seed,
+        auth_scheme,
+        auth_seed,
+    };
+
+    let json_data = serde_json::to_string(&account_data)?;
+
+    fs::write(file_path.as_path(), json_data).map_err(|err| {
+        anyhow!("Failed to write account file to {}, Error: {err}", file_path.display())
+    })?;
+
+    Ok(())
+}
+
+fn create_accounts(accounts: &[AccountInput]) -> Result<Vec<Account>> {
+    // create the accounts folder
+    let accounts_folder_path = PathBuf::from(DEFAULT_ACCOUNTS_FOLDER);
+
+    fs::create_dir(accounts_folder_path.as_path())
+        .map_err(|err| anyhow!("Failed to create accounts folder: {err}"))?;
+
+    let mut final_accounts = Vec::new();
+
+    for account in accounts {
+        match account {
+            AccountInput::BasicWallet(inputs) => {
+                println!("Generating basic wallet account... ");
+                let seed: [u8; 32] = hex_string_to_byte_array(&inputs.seed)?;
+                let auth_seed: [u8; 40] = hex_string_to_byte_array(&inputs.auth_seed)?;
+
+                let keypair = KeyPair::from_seed(&auth_seed)?;
```

Comment:
I tried not to create a complex data type for Auth because I am not sure where it would live. 

Would we add it to miden_lib in this file: 
https://github.com/0xPolygonMiden/miden-base/blob/1841b99c776722854444669cc2168e2eed49bac2/miden-lib/src/auth.rs#L4-L10

Reply:
And it seems that it would add many new types, also not considering the *input ones: 
- AuthScheme (already exists)
- AuthInput
- AuthInfo

Should we find a better way to handle this or does that seem okay for you?

Reply:
Yep - we may need to find a better way, but I'd probably leave this normalization for the PR when we move these structs into `miden-base`.